pypeit/display/ginga_spec1dview.py

```diff
@@ -0,0 +1,482 @@
+        """
+        self.masked = self.masked_options[idx]
+        self.logger.debug(f"Selected masked option: {self.masked}")
+        self.recalc()
+
+    def plot_lines(self):
+        """Plot the line list.
+
+        Lines are made into a single compound object so that it is easier
+        to remove them as a group if the line list is changed.
+        """
+        canvas = self.plot.get_canvas()
+        canvas.delete_object_by_tag('lines', redraw=False)
+
+        lines = self.dc.Canvas()
+        if self.llist is not None:
+            # NOTE: adapted straight from linetools
+            x_min, x_max = self.data.x_min, self.data.x_max
+            y_min, y_max = self.data.y_min, self.data.y_max
+
+            z = self.z
+            wvobs = np.array((1 + z) * self.llist.wrest)
+            ylbl_pos = y_max - 0.2 * (y_max - y_min)
+            gdwv = np.where((wvobs > x_min) & (wvobs < x_max))[0]
+
+            for kk in range(len(gdwv)):
+                jj = gdwv[kk]
+                wrest = self.llist.wrest[jj].value
+                lbl = self.llist.name[jj]
+                # Plot
+                x_data = wrest * np.array([z + 1, z + 1])
+                y_data = (y_min, y_max)
+                lines.add(self.dc.Line(x_data[0], y_data[0],
+                                       x_data[1], y_data[1],
+                                       linewidth=1,
+                                       #linestyle='dotted',
+                                       linestyle='solid',
+                                       color='blue'), redraw=False)
+                # Label
+                x, y = wrest * (z + 1), ylbl_pos
+                lines.add(self.dc.Text(x, y, text=lbl, rot_deg=90,
+                                       color='blue', fontsize=10),
+                          redraw=False)
+
+        # will be empty if there were no lines
+        canvas.add(lines, tag='lines', redraw=False)
+
+        # this causes the plot viewer to redraw itself
+        self.plot.make_callback('modified')
+
+    def replot(self):
+        """Replot the plot and line list.
+        """
+        if self.plot is None:
+            return
+        # plot flux vs. wavelength
+        self.plot.clear()
+        canvas = self.plot.get_canvas()
+
+        # plot flux vs. wavelen
+        points = np.array((self.data.wave, self.data.flux)).T
+        p1 = self.dc.Path(points, linewidth=1, linestyle='solid',
+                          alpha=0.7, color='black')
+        canvas.add(p1, tag='spectrum', redraw=False)
+
+        # optionally, plot error vs. wavelength
+        if self.settings.get('plot_error', False):
+            points = np.array((self.data.wave, self.data.sig)).T
+            p2 = self.dc.Path(points, linewidth=2, linestyle='solid',
+                              alpha=1.0, color='red')
+            canvas.add(p2, tag='error', redraw=False)
+
+        self.plot.set_titles(title="Spec1dView",
```

Comment:
Is there value in setting this title to something that matches the filename/extension being plotted, rather than a static Spec1dViewer?